vocode/streaming/utils/setup_tracer.py

```diff
@@ -0,0 +1,62 @@
+# vocode/streaming/utils/setup_tracer.py
+
+import os
+from typing import Optional
+
+from opentelemetry import trace
+from opentelemetry.exporter.cloud_trace import CloudTraceSpanExporter
+from opentelemetry.sdk.trace import TracerProvider
+from opentelemetry.sdk.trace.export import BatchSpanProcessor, ConsoleSpanExporter
+from opentelemetry.trace import Span, Status, StatusCode
+
+tracer = trace.get_tracer(__name__)
+
+
+def setup_tracer():
+    """
+    Setup the tracer for the application based on the environment.
+    Environment variable ENV should be set to "prod" for production environment and "dev" for development environment.
+    """
+    try:
+        tracer_provider = TracerProvider()
+        project_id = os.getenv("GOOGLE_CLOUD_PROJECT", "default_project_id")
+        cloud_trace_exporter = CloudTraceSpanExporter(project_id=project_id)
+        if os.getenv("ENV") == "prod":
```

Comment:
did you run your test in prod mode? It looks like this code skips some of the setup in dev

I'm a bit worried because we _have_ to test this in prod. There's simply no way to guarantee that our laptops have the same auth environment as porter. So I want to be sure that if this runs in prod, and tracing permissions are broken, we don't get a SEV

Reply:
i ran it in dev first

Reply:
does that mean you ran the PR as-is in a dev environment? or that you ran the code I'm asking in dev (i.e. removed the if statement for testing)?

ultimately, what I really want to know is if this could break prod